src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2092,12 +2104,28 @@ public override void AfterTransactionBegin(ITransaction tx)
 			using (new SessionIdLoggingContext(SessionId))
 			{
 				CheckAndUpdateSessionStatus();
-				Interceptor.AfterTransactionBegin(tx);
+
+				if (!_transactionCoordinatorShared)
+				{
+					Interceptor.AfterTransactionBegin(tx);
+				}
```

Comment:
Bit muddy subject here.

In the case of `AdoTransaction`, the transaction events were not called on sharing sessions, so this check keeps that behavior for the interceptor.

In the case of transaction scope, each session was still enlisting itself, and was receiving events. So the interceptor was call for each sharing session and for the originating one, excepted for the after transaction event which was already guarded for being fired only from originating session... I have decided to align on that behavior for all events.

Now, why not firing the call on the sharing session interceptor? Maybe because by default, it is the same than the originating session. So it would get notified multiples times, likely causing issues. Maybe we should check if it is the same, and if not, then firing the events. Going to add that change.